* Fix new Sonar issue in the `AbstractMailReceiver`

artembilan opened the pull request:
<!--
Thanks for contributing to Spring Integration. Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #).

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc).
-->

artembilan committed:
* Fix new Sonar issue in the `AbstractMailReceiver`

garyrussell commented:
This is still a problem

```java
		finally {
			MailTransportUtils.closeFolder(folder, this.shouldDeleteMessages);
			this.folderReadLock.unlock();
		}
```

We won't unlock if the close() fails.

artembilan committed:
* Move `unlock()` to its own `finally`

artembilan committed:
* Make `ImapMailReceiverTests.imapIdleServer` non-static to allow
recreate the mail server for each test method

https://build.spring.io/browse/INT-MASTERSPRING40-591/

artembilan committed:
* Remove MongoDb download from Travis config

artembilan committed:
* Fix Checkstyle violation
